lib/java/spring/auth/src/main/java/co/airy/spring/auth/oidc/UserService.java

```diff
@@ -0,0 +1,52 @@
+package co.airy.spring.auth.oidc;
+
+import co.airy.spring.auth.oidc.github.EmailsResponse;
+import co.airy.spring.auth.oidc.github.GithubApi;
+import org.springframework.security.oauth2.client.userinfo.DefaultOAuth2UserService;
+import org.springframework.security.oauth2.client.userinfo.OAuth2UserRequest;
+import org.springframework.security.oauth2.client.userinfo.OAuth2UserService;
+import org.springframework.security.oauth2.core.OAuth2AuthenticationException;
+import org.springframework.security.oauth2.core.OAuth2Error;
+import org.springframework.security.oauth2.core.OAuth2ErrorCodes;
+import org.springframework.security.oauth2.core.user.DefaultOAuth2User;
+import org.springframework.security.oauth2.core.user.OAuth2User;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.stream.Collectors.toList;
+
+public class UserService implements OAuth2UserService<OAuth2UserRequest, OAuth2User> {
+    private final GithubApi githubApi;
+    private final DefaultOAuth2UserService userService = new DefaultOAuth2UserService();
+
+    public UserService(GithubApi githubApi) {
+        this.githubApi = githubApi;
+    }
+
+    public OAuth2User loadUser(OAuth2UserRequest userRequest) throws OAuth2AuthenticationException {
+        OAuth2User user = userService.loadUser(userRequest);
+
+        // Github does not return a user's email by default
+        if ("github".equals(userRequest.getClientRegistration().getRegistrationId())) {
```

Comment:
GitHub does not actually comply with OIDC. Dex uses the same approach of requesting the email scope and then fetching them separately to fake oidc authentication.

Therefore we create a custom user service that does this extra lookup